### PR TITLE
Update AppLifecycle Activation.md

### PR DESCRIPTION
**Pull request by andreww-msft**
Added the exePath parameter to Register/Unregister methods.

### CODE REVIEW

**Comment by riverar** [comment]
Missing documentation on what constitutes a valid `image resource file`.

File: specs/AppLifecycle/Activation/AppLifecycle Activation.md [hunk 1]
```diff
@@ -245,8 +252,9 @@ unregister for these activation kinds.
 void UnregisterForActivation()
 {
     // Unregister one or more registered filetypes.
+    string imageFileTypes[3] = { ".jpg", ".png", ".bmp" };
     ActivationRegistrationManager::UnregisterForFileTypeActivation(
-        new string[]{ ".jpg", ".png", ".bmp" });
+        imageFileTypes);
 
     ActivationRegistrationManager::UnregisterForProtocolActivation("foo");
     ActivationRegistrationManager::UnregisterForStartupActivation();
```
Comment (aeloros) [inline specs/AppLifecycle/Activation/AppLifecycle Activation.md:260]: Don't we need to specify which taskId we're unregistering?

File: specs/AppLifecycle/Activation/AppLifecycle Activation.md [hunk 2]
```diff
@@ -161,18 +161,25 @@ void RegisterForActivation()
...
     // and zero or more verbs for the File Explorer context menu.
+    string imageFileTypes[3] = { ".jpg", ".png", ".bmp" };
+    string verbs[2] = { "view", "edit" };
     ActivationRegistrationManager::RegisterForFileTypeActivation(
-        new string[]{ ".jpg", ".png", ".bmp" },
-        "MyResources.dll",
-        12,
-        new string[]{ "view", "edit" });
+        imageFileTypes,
+        "MyResources.dll", 0,
```
Comment (aeloros) [inline specs/AppLifecycle/Activation/AppLifecycle Activation.md:168]: nit: I would pick one style or another here for line-wrapping parameters.  Either one per line, or wrap using 'word' style wrapping at a specific point.

File: specs/AppLifecycle/Activation/AppLifecycle Activation.md [hunk 3]
```diff
@@ -161,18 +161,25 @@ void RegisterForActivation()
...
+
+    // Register another set, and specifying the EXE path.
+    string beerFileTypes[1] = { ".ipa" };
+    ActivationRegistrationManager::RegisterForFileTypeActivation(
+        beerFileTypes,
+        "MyResources.dll", 34,
+        "C:\\Program Files\\Contoso\\MyApp.exe");
 
     // Register some URI schemes for protocol activation.
     ActivationRegistrationManager::RegisterForProtocolActivation("foo");
```
Comment (aeloros) [inline specs/AppLifecycle/Activation/AppLifecycle Activation.md:185]: Does protocol not have a version that takes an EXE?

File: specs/AppLifecycle/Activation/AppLifecycle Activation.md [hunk 4]
```diff
@@ -161,18 +161,25 @@ void RegisterForActivation()
...
+        imageFileTypes,
+        "MyResources.dll", 0,
+        verbs);
 
-    // Register another set of filetypes, with a different
-    // icon, and no specified verbs.
+    // Register another set of filetypes, falling back to the
+    // default icon, and no specified verbs.
+    string videoFileTypes[3] = { ".mov", ".wmv", ".mp3" };
     ActivationRegistrationManager::RegisterForFileTypeActivation(
```
Comment (aeloros) [inline specs/AppLifecycle/Activation/AppLifecycle Activation.md:174]: Where are the Startup Task register calls?

File: specs/AppLifecycle/Activation/AppLifecycle Activation.md [hunk 5]
```diff
@@ -344,23 +352,35 @@ Similarly, if the app registers multiple times for startup, this is also
...
+| API                                                          | Description                                                  |
+| ------------------------------------------------------------ | ------------------------------------------------------------ |
+| static void UnregisterForFileTypeActivation(<br> string fileTypes[], <br> string exePath<br> ); | **fileTypes**: one or more filetypes that the caller wants to unregister. <br>**exePath**: path to the executable that was registered for activation. |
+| static void UnregisterForFileTypeActivation(<br> string fileTypes[]<br> ); |                                                              |
+| static void UnregisterForProtocolActivation(<br> string name,<br> string exePath<br> ); | **name**: the previously-registered protocol identifier, eg "foo" or "https". <br>**exePath**: path to the executable that was registered for activation. |
+| static void UnregisterForProtocolActivation(<br> string name<br> ); |                                                              |
+| static void UnregisterForStartupActivation(<br> string exePath<br>); | **exePath**: path to the executable that was registered for activation. |
```
Comment (aeloros) [inline specs/AppLifecycle/Activation/AppLifecycle Activation.md:382]: I don't think this one makes sense.  Unregister (in all of these cases) I *think* should be based on the type and not the EXE.  So fileTypes[], protocol Scheme 'name', and Startup's taskId.

File: specs/AppLifecycle/Activation/AppLifecycle Activation.md [hunk 6]
```diff
@@ -344,23 +352,35 @@ Similarly, if the app registers multiple times for startup, this is also
...
+| static void RegisterForProtocolActivation(<br/> string name,<br/> string exePath<br/>); | **name**:<br/>the protocol identifier, eg "foo" or "https".<br/> **exePath**: path to the<br/>executable to be activated. |
+| static void RegisterForProtocolActivation(<br/> string name<br/> ); |                                                              |
+| static void RegisterForStartupActivation(<br/><br/> string taskId,<br/> bool isDisabled,<br/> string displayName,<br/> string exePath); | **taskId**: a string identifier which the app can use in<br/>the StartupTask API to request enabling or disabling of the startup<br/>behavior.<br/> **isDisabled**: the startup behavior defaults to isDisabled=true. Unpackaged apps can enable the behavior through this API. Centennial apps can also enable it,<br/> so long as the user has activated the app at least once. UWP apps cannot directly enable this - the user has to go to Settings to enable it. These are the 3 existing behaviors for the different app types.<br/> **displayName**: used in the Settings -<br/>Apps - Startup page.<br/> **exePath**: path to the executable to be activated. |
```
Comment (aeloros) [inline specs/AppLifecycle/Activation/AppLifecycle Activation.md:366]: DisplayName is not controllable through the API.  The platform takes it from the information embedded in the version resource data for the .EXE target.  We need to remove this from the parameters.

**Review by aeloros — CHANGES_REQUESTED** [review]

**Review by aeloros — APPROVED** [review]

File: specs/AppLifecycle/Activation/AppLifecycle Activation.md (hunk 5 above)
Comment (aeloros) [reply specs/AppLifecycle/Activation/AppLifecycle Activation.md:382]: We talked offline, and Unregister still needs the exePath for File and Protocol but not Startup.  Instead it needs taskId.

File: specs/AppLifecycle/Activation/AppLifecycle Activation.md (hunk 4 above)
Comment (andreww-msft) [reply specs/AppLifecycle/Activation/AppLifecycle Activation.md:175]: Further down in the doc.

File: specs/AppLifecycle/Activation/AppLifecycle Activation.md (hunk 3 above)
Comment (andreww-msft) [reply specs/AppLifecycle/Activation/AppLifecycle Activation.md:187]: Yes, defined below.

File: specs/AppLifecycle/Activation/AppLifecycle Activation.md (hunk 1 above)
Comment (andreww-msft) [reply specs/AppLifecycle/Activation/AppLifecycle Activation.md]: Fixed.

File: specs/AppLifecycle/Activation/AppLifecycle Activation.md (hunk 5 above)
Comment (andreww-msft) [reply specs/AppLifecycle/Activation/AppLifecycle Activation.md]: Fixed.

File: specs/AppLifecycle/Activation/AppLifecycle Activation.md (hunk 6 above)
Comment (andreww-msft) [reply specs/AppLifecycle/Activation/AppLifecycle Activation.md]: Fixed.

File: specs/AppLifecycle/Activation/AppLifecycle Activation.md (hunk 2 above)
Comment (andreww-msft) [reply specs/AppLifecycle/Activation/AppLifecycle Activation.md]: Fixed.

File: specs/AppLifecycle/Activation/AppLifecycle Activation.md [hunk 7]
```diff
@@ -344,23 +352,35 @@ Similarly, if the app registers multiple times for startup, this is also
...
-| static void UnregisterForStartupActivation();                              |                                                                               |
+| API                                                          | Description                                                  |
+| ------------------------------------------------------------ | ------------------------------------------------------------ |
+| static void UnregisterForFileTypeActivation(<br> string fileTypes[], <br> string exePath<br> ); | **fileTypes**: one or more filetypes that the caller wants to unregister. <br>**exePath**: path to the executable that was registered for activation. |
+| static void UnregisterForFileTypeActivation(<br> string fileTypes[]<br> ); |                                                              |
+| static void UnregisterForProtocolActivation(<br> string name,<br> string exePath<br> ); | **name**: the previously-registered protocol identifier, eg "foo" or "https". <br>**exePath**: path to the executable that was registered for activation. |
+| static void UnregisterForProtocolActivation(<br> string name<br> ); |                                                              |
```
Comment (aeloros) [inline specs/AppLifecycle/Activation/AppLifecycle Activation.md:381]: This should be either scheme or schemeName.

File: specs/AppLifecycle/Activation/AppLifecycle Activation.md [hunk 8]
```diff
@@ -344,23 +352,35 @@ Similarly, if the app registers multiple times for startup, this is also
...
-| static void UnregisterForStartupActivation();                              |                                                                               |
+| API                                                          | Description                                                  |
+| ------------------------------------------------------------ | ------------------------------------------------------------ |
+| static void UnregisterForFileTypeActivation(<br> string fileTypes[], <br> string exePath<br> ); | **fileTypes**: one or more filetypes that the caller wants to unregister. <br>**exePath**: path to the executable that was registered for activation. |
+| static void UnregisterForFileTypeActivation(<br> string fileTypes[]<br> ); |                                                              |
+| static void UnregisterForProtocolActivation(<br> string name,<br> string exePath<br> ); | **name**: the previously-registered protocol identifier, eg "foo" or "https". <br>**exePath**: path to the executable that was registered for activation. |
```
Comment (aeloros) [inline specs/AppLifecycle/Activation/AppLifecycle Activation.md:380]: Why is there no logo support for Protocol?

File: specs/AppLifecycle/Activation/AppLifecycle Activation.md [hunk 9]
```diff
@@ -344,23 +352,35 @@ Similarly, if the app registers multiple times for startup, this is also
...
+| static void<br/>RegisterForFileTypeActivation(<br/> string supportedFileTypes[],<br/> string logo,<br/> int index,<br/> string exePath<br/>); |                                                              |
```
Comment (aeloros) [inline specs/AppLifecycle/Activation/AppLifecycle Activation.md:359]: For File and Protocol both we have the ability to specify a DisplayName which would show up in the OpenWith UI.  I'm not sure what the behavior is if we don't set it.  I'll figure that out, but if it's a bad experience we should consider adding it to these ones.

Comment (andreww-msft) [reply specs/AppLifecycle/Activation/AppLifecycle Activation.md:359]: Fixed.

File: specs/AppLifecycle/Activation/AppLifecycle Activation.md (hunk 8 above)
Comment (andreww-msft) [reply specs/AppLifecycle/Activation/AppLifecycle Activation.md:380]: Fixed.

File: specs/AppLifecycle/Activation/AppLifecycle Activation.md (hunk 7 above)
Comment (andreww-msft) [reply specs/AppLifecycle/Activation/AppLifecycle Activation.md:381]: Fixed.

**Pull request merged**